### ISSUE TITLE
Validate email addresses sent into offline downloads against userdetails

### ISSUE THREAD

**Issue by ansell** [AtlasOfLivingAustralia/biocache-service#479]
It would be valuable to validate the email addresses sent into offline downloads against userdetails.

This is in addition to the changes in #474 (the current fix for #343 ) which only check for valid email address syntax, but don't match against our registered user list.

**Comment by matthewandrews** [comment]
Specifically, a check against userdetails should test that the account is activated and is unlocked (i.e. email address is confirmed).

**Comment by brucehyslop** [comment]
The properties available from the auth web service are `locked` and `roles`.

Will check that the user details are not locked, a update would need to be made to the auth service to return the `activated` status. 

Should we also check that the user is a member of role `ROLE_USER`?

**Comment by ansell** [comment]
@brucehyslop At this point, checking against `ROLE_USER` should be a trivial check, but may still be useful to setup using a configurable role name in-case other Living Atlases want to restrict downloads on their sites to particular roles. Currently we never remove `ROLE_USER`, we solely rely on the `locked` attribute to stop logins.

I have opened a ticket on userdetails to add the activated attribute to the web service result https://github.com/AtlasOfLivingAustralia/userdetails/issues/94

**Comment by nickdos** [comment]
Is this ready for testing or waiting for the change in userdetails (dependency)?

**Comment by ansell** [comment]
It is waiting on merging this PR: https://github.com/AtlasOfLivingAustralia/userdetails/pull/95 and then deploying the userdetails snapshot to auth-test

**Comment by Rita-C** [comment]
Bruce was working on this issue, will it make 2.5.0 release @brucehyslop ?

**Comment by brucehyslop** [comment]
There are a couple of requested code changes that I'll make today prior to merging of PR #520

**Comment by brucehyslop** [comment]
PR #520 has been merged and develop branch deployed to test.

**Comment by Rita-C** [comment]
Tested:
1. an offline download with a valid but not registered email address
2. an offline download with a registered email but the account was locked and not active
Both received 403 as expected.

**Comment by ansell** [comment]
Opening a new issue for providing backwards compatibility with previous userdetails versions in #540 and closing this issue as complete.

**Comment by ansell** [comment]
Reopening this issue as the deployment was rolled back.

**Comment by sylmorin-gbif** [comment]
Hi,
With this change, the download is no more possible if CAS service is not activated.
I agree that it's better to use it (and I will add it soon), but it's important to state that CAS is no more "optional" now.
Regards,
Sylvain

**Comment by nickdos** [comment]
@sylmorin-gbif I agree it should be optional - see new issue #546.
